crates/ruff_linter/src/rules/pylint/rules/global_at_module_level.rs

```diff
@@ -0,0 +1,34 @@
+use ast::Stmt;
+use ruff_python_ast::{self as ast};
+
+use ruff_diagnostics::{Diagnostic, Violation};
+use ruff_macros::{derive_message_formats, violation};
+use ruff_text_size::Ranged;
+
+use crate::checkers::ast::Checker;
+
+/// ## What it does
+/// Checks for the use of `global` at the top level of a module.
+///
+/// ## Why is this bad?
+/// `global` at the top level of a module is redundant.
+#[violation]
+pub struct GlobalAtModuleLevel;
+
+impl Violation for GlobalAtModuleLevel {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        format!("`global` at module level is redundant")
+    }
+}
+
+/// PLW0604
+pub(crate) fn global_at_module_level(checker: &mut Checker, stmt: &Stmt) {
+    if !checker.semantic().at_top_level() {
+        return;
+    }
```

Comment:
This won't catch cases like:

```python
if True:
    global X
```

Should those be flagged? If so, we should use `checker.semantic().current_scope().kind.is_module()`.

Reply:
Good catch! I do think they should.